Scripts/BuildGVFSForWindows.bat

```diff
@@ -39,4 +39,6 @@ IF NOT EXIST %msbuild% (
 
 %msbuild% %VFS_SRCDIR%\GVFS.sln /p:GVFSVersion=%GVFSVersion% /p:Configuration=%SolutionConfiguration% /p:Platform=x64 || exit /b 1
 
+SET SolutionDir=%VFS_SRCDIR%\

```

Comment:
This seems odd, but the .NET Core version of MSBuild is failing to resolve BuildOutputDir locations correctly without this change:
```
CSC : error CS2001: Source file 'C:\Users\nickgra\VFSForGit..\BuildOutput\CommonAssemblyVersion.cs' could not be found. [C:\Users\nickgra\VFSForGit\GVFS\GVFS.GVFlt\GVFS.GVFlt.csproj]
```

I think this change will be safe to make in the top level Initialize script, but I'll dig in deeper here.

Reply:
Your explanation is correct. I fixed this and chose to pass SolutionDir into `dotnet` since that is technically the missing variable (once it's defined, BuildOutputDir and friends are all defined correctly per our props files).
Publishing the solution at this time was getting complicated due to some of our custom MSBuild logic in GVFS.Upgrader. I'd prefer to address that as part of a future PR. I know saying that is what caused our bug with `dotnet restore`, but I'd like to rationalize all of our .NET Core stuff sooner rather than later.